### ISSUE TITLE
Choosing a kernel for the first time does not cause a kernel change event

### ISSUE THREAD

**Issue by imadjarov-quantinuum** [microsoft/vscode-jupyter#16112]
### Applies To

- [X] Notebooks (.ipynb files)
- [ ] Interactive Window and\/or Cell Scripts (.py files with \#%% markers)

### What happened?

This is related to the recently closed https://github.com/microsoft/vscode-jupyter/pull/15967  (@rchiodo @DonJayamanne)

When choosing a kernel for a notebook that has had no previous kernel selected, a "kernel change event" (as discussed in the issue above) is not fired. For example, Pylance (v2024.10.1) does not start highlighting syntax with the newly selected kernel. Only once another kernel is selected again does this event fire and Pylance starts highlighting code. It seems like the above bugfix only works once you select a kernel more than once. 

I am running this via SSH Remote btw.

### VS Code Version

1.94.0

### Jupyter Extension Version

2024.9.1

### Coding Language and Runtime Version

Python

### Language Extension Version (if applicable)

2024.16.0

### Running Jupyter locally or remotely?

Remote

**Comment by DonJayamanne** [comment]
@rchiodo I'm struggling to understand this issue here. If no kernel was ever selected, then selecting a kernel will definitely cause a change event as there was no information to start with.

@imadjarov-quantinuum Please can you install the latest pre-release versions of Jupyter and Pylance and confirm this works.

**Comment by rchiodo** [comment]
@DonJayamanne I haven't seen this myself. Does jupyter log the get kernel request? Pylance does not, it just makes the call when getting the python path for a file.

**Comment by imadjarov-quantinuum** [comment]
> [@rchiodo](https://github.com/rchiodo) I'm struggling to understand this issue here. If no kernel was ever selected, then selecting a kernel will definitely cause a change event as there was no information to start with.
> 
> [@imadjarov-quantinuum](https://github.com/imadjarov-quantinuum) Please can you install the latest pre-release versions of Jupyter and Pylance and confirm this works.

I upgraded to VS Code 1.95.1, and the pre-release versions of both Jupyter and Pylance, and the issue persists.

**Comment by DonJayamanne** [comment]
@rchiodo I think its time to move to a more robust approach to fixing this issue.
Closing in favour of https://github.com/microsoft/vscode-jupyter/issues/15987
